packages/contracts/mainnetDeployment/balancerPoolDeployment.js

```diff
@@ -0,0 +1,154 @@
+// Test with:
+// GAS_PRICE=0 BLOCK_NUMBER=12612689 npx hardhat run mainnetDeployment/balancerPoolDeployment.js --config hardhat.config.mainnet-fork.js
+
+const { WeightedPool2TokensFactory } = require("./ABIs/WeightedPool2TokensFactory.js")
+const { WeightedPool2Tokens } = require("./ABIs/WeightedPool2Tokens.js")
+const { IVault } = require("./ABIs/IVault.js")
+const { ERC20 } = require("./ABIs/ERC20.js")
+const { WETH: WETH_ABI } = require("./ABIs/WETH.js")
+const toBigNum = ethers.BigNumber.from
+const { TestHelper: th, TimeValues: timeVals } = require("../utils/testHelpers.js")
+const { dec } = th
+// Addresses are the same on all networks
+
+const VAULT = '0xBA12222222228d8Ba445958a75a0704d566BF2C8';
+
+const WEIGHTED_POOL_FACTORY = '0x8E9aa87E45e92bad84D5F8DD1bff34Fb92637dE9';
+const ORACLE_POOL_FACTORY = '0xA5bf2ddF098bb0Ef6d120C98217dD6B141c74EE0';
+
+const DELEGATE_OWNER = '0xBA1BA1ba1BA1bA1bA1Ba1BA1ba1BA1bA1ba1ba1B';
+
+// Mainnet addresses; adjust for testnets
+
+const WETH = '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2';
+const LUSD = '0x5f98805A4E8be255a32880FDeC7F6728C6568bA0';
+const PRICE_FEED = '0x4c517D4e2C851CA76d7eC94B805269Df0f2201De';
+
+const tokens = [LUSD, WETH];
+const weights = [toBigNum(dec(4, 17)), toBigNum(dec(6, 17))];
+
+const NAME = 'WETH/LUSD Pool';
+const SYMBOL = '60WETH-40LUSD';
+const swapFeePercentage = toBigNum(dec(5, 15)); // 0.5%
+
+const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
+
+const INITIAL_FUNDING = toBigNum(dec(5, 22)); // $50k
+
+async function main() {
+  // Uncomment for testing:
+
+  const impersonateAddress = "0x787EfF01c9FdC1918d1AA6eeFf089B191e2922E4"
+  await hre.network.provider.request({
+    method: "hardhat_impersonateAccount",
+    params: [ impersonateAddress ]
+  })
+  const deployerWallet = await ethers.provider.getSigner(impersonateAddress)
+  const deployerWalletAddress = impersonateAddress
+
+  //const deployerWallet = (await ethers.getSigners())[0]
+  //const deployerWalletAddress = deployerWallet.address
+
+  const factory = new ethers.Contract(
+    ORACLE_POOL_FACTORY,
+    WeightedPool2TokensFactory.abi,
+    deployerWallet
+  );
+  const vault = new ethers.Contract(
+    VAULT,
+    IVault.abi,
+    deployerWallet
+  );
+
+  const priceFeed = await ethers.getContractAt('PriceFeed', PRICE_FEED)
+
+  // ZERO_ADDRESS owner means fixed swap fees
+  // DELEGATE_OWNER grants permission to governance for dynamic fee management
+  // Any other address lets that address directly set the fees
+  const oracleEnabled = true;
+  const tx1 = await factory.create(
+    NAME, SYMBOL, tokens, weights,
+    swapFeePercentage, oracleEnabled,
+    //DELEGATE_OWNER
+    impersonateAddress
```

Comment:
and I guess change this `impersonateAddress` for real mainnet deployment? Who should have control over fee-setting for our pool?

Reply:
Ha, good catch! In spite of adding this comment:
```
// Uncomment for testing:
```
I forgot... :grimacing: